evm_loader/program/src/instruction/transaction_cancel.rs

```diff
@@ -53,17 +53,16 @@ fn execute<'a>(
 ) -> Result<()> {
     let trx_chain_id = storage.trx().chain_id().unwrap_or(DEFAULT_CHAIN_ID);
 
-    let used_gas = U256::ZERO;
-    let total_used_gas = storage.gas_used();
+    let used_gas = U256::from(CANCEL_TRX_COST + LAST_ITERATION_COST);
+    let total_used_gas = storage.gas_used() + used_gas;
     log_data(&[
         b"GAS",
         &used_gas.to_le_bytes(),
         &total_used_gas.to_le_bytes(),
     ]);
 
-    let gas = U256::from(CANCEL_TRX_COST + LAST_ITERATION_COST);
-    let priority_fee = priority_fee_txn_calculator::handle_priority_fee(storage.trx(), gas)?;
-    let _ = storage.consume_gas(gas, priority_fee, accounts.try_operator_balance()); // ignore error
+    let priority_fee = priority_fee_txn_calculator::handle_priority_fee(storage.trx(), used_gas)?;
+    let _ = storage.consume_gas(used_gas, priority_fee, accounts.try_operator_balance()); // ignore error
```

Comment:
It is possible that there is not enough gas available to pay `CANCEL_TRX_COST + LAST_ITERATION_COST`, in which case the actual gas used will be less.

Reply:
Hm, I think if there's not enough gas available, the `storage.consume_gas` will throw `OutOfGas` error (as it should do).

Therefore, the logic is accurate and we either spend 10k of gas (and do `log_data(GAS, 10k, total_gas_used)`) or we throw the error.

Did you mean we should change the logic and charge less than 10k (in case total_gas goes beyond the limit)?
IMO, such cases should be handled by improving `estimate_gas` and overall estimation process on the proxy/api side?